### ISSUE TITLE
Docs: highlight full method for '[source]' link

### ISSUE THREAD

**Issue by sadielbartholomew** [NCAS-CMS/cf-python#51]
Instead of linking just to the first line of the relevant method from a '[source]' link in the API reference of the documentation, it would be better to link to the full method. In other words, the link would go to a page for the relevant module in the codebase and have highlighted multiple lines covering the extent of the method, rather than just the one with the ``def`` declaring it.

The aspiration to do this was noted in a comment in the relevant code , https://github.com/NCAS-CMS/cf-python/commit/00b2c7992d35c44479dc785947a63547e7931c14#diff-1ec61223727a10e5e4c74c68336809fcL474 which I removed, as there is much greater visibility here in the Issue Tracker. The comment implies it was not possible to link to a range in BitBucket but in GitHub it is simple: e.g. https://github.com/NCAS-CMS/cf-python/blob/master/docs/source/conf.py#L464-L468. In fact, there is a variable ``nlines`` which looks like it provides the number of code lines for a method so it will be trivial to add that to the number of the first line of the method to get the final line and link from the first to that in the ``#L{}-{}`` pattern after the domain.

This is not urgent but should be straightforward so ideally we can address it for the next release.

**Comment by davidhassell** [comment]
This would be great - thanks for raising it.

**Comment by sadielbartholomew** [comment]
Solved in 920e573217aaa37feb8102b85aee9ac7a946f121. I have not pushed any rebuilt docs yet as there API changes have since gone in so we should wait until the release of v3.3.0, but I tested it as working by building locally with the ``dev`` arg & changing ``blob/v3.3.0`` to ``blob/v3.2.0`` to see where a sample of the source code links went. Random example:

![scr_linkcode](https://user-images.githubusercontent.com/30274190/79249555-57f26c00-7e75-11ea-86ec-5aa28df278f9.png)

All line number ranges were correct, except in some cases where they shifted but only because of code changes making line numbers for the methods/classes in question differ between 3.2.0 & 3.3.0. Cross-referencing onto the current codebase showed these were generated correctly for that.

``# --- End: class`` markers aren't strictly part of classes so don't included, but IMO that's not worth effort to solve.
![scr_linkcode_2](https://user-images.githubusercontent.com/30274190/79249552-5759d580-7e75-11ea-9c86-334f015e8dd5.png)
